spec/_plugins/alias_generator_spec.rb

```diff
@@ -28,7 +28,35 @@
       })
     
     author_index = Jekyll::AuthorIndex.new(site, "base", "author_dir", "Author") 
-    expect(author_index.data).to eq({ 'author' => "Author", 'description' => 'author: Author', "publications" => [video, post], 'title' => "Author"})
+    
+    expect(author_index.data['author']).to eq("Author")
+    expect(author_index.data['description']).to eq("author: Author")
+    expect(author_index.data['publications']).to include(video, post)
+    expect(author_index.data['publications']).not_to include(newsletter)
+    expect(author_index.data['title']).to eq("Author")
+  end
+
+  it 'Author\'s publications are sorted descendingly by date' do
+    site = double
+    file = double
+    early_post = double
+    late_post = double
+
+    allow(Jekyll::Utils).to receive(:merged_file_read_opts)
+    allow(File).to receive(:read).and_return(file)
+
+    allow(site).to receive(:in_source_dir)
+    allow(site).to receive(:file_read_opts)
+    allow(site).to receive(:config).and_return({})
+    
+    allow(early_post).to receive(:data).and_return({ 'author' => "Author", 'title' => "", 'date' => Date.new(2000,1,1) })
+    allow(late_post).to receive(:data).and_return({ 'author' => "Author", 'title' => "", 'date' => Date.new(2001,1,1) })
```

Comment:
The "early" and "late" naming confused me a bit; "older" and "newer" may be better. But, that could just be me.

Reply:
Nice suggestion!